xchange-bittrex/src/main/java/org/knowm/xchange/bittrex/BittrexAdapters.java

```diff
@@ -219,14 +223,61 @@ public static Wallet adaptWallet(Collection<BittrexBalance> balances) {
     return Wallet.Builder.from(wallets).build();
   }
 
-  public static void adaptMetaData(List<BittrexSymbol> rawSymbols, ExchangeMetaData metaData) {
-    BittrexAdapters.adaptCurrencyPairs(rawSymbols)
-        .forEach(
-            currencyPair -> {
-              metaData.getCurrencyPairs().putIfAbsent(currencyPair, null);
-              metaData.getCurrencies().putIfAbsent(currencyPair.base, null);
-              metaData.getCurrencies().putIfAbsent(currencyPair.counter, null);
-            });
+  public static void adaptMetaData(List<BittrexSymbol> rawSymbols,
+                                   List<BittrexCurrency> bittrexCurrencies,
+                                   Map<CurrencyPair, Fee> dynamicTradingFees,
+                                   ExchangeMetaData metaData) {
+    List<CurrencyPair> currencyPairs = BittrexAdapters.adaptCurrencyPairs(rawSymbols);
+    for (CurrencyPair currencyPair : currencyPairs) {
+      CurrencyPairMetaData defaultCurrencyPairMetaData = metaData.getCurrencyPairs().get(currencyPair);
+      BigDecimal resultingFee = null;
+      // Prioritize dynamic fee
+      if( dynamicTradingFees != null ){
+        Fee fee = dynamicTradingFees.get(currencyPair);
+        if( fee != null){
+          resultingFee = fee.getMakerFee();
+        }
+      }else{
```

Comment:
your formatting is still wrong in many places which indicates that you are not using maven to build this project

use maven - it will autoformat your source code

Reply:
I use maven, it is not possible to build without maven !!!!

Reply:
Formatted this and the other changed lines in the source files.

Reply:
thank you for fixing the formatting

ps. Eclipse and other IDEs for a long time used their own build tools instead of invoking maven to build maven projects - I don't know how Eclipse behaves now but having unformatted code was a obvious symptom of this as the IDE build tools omitted the maven plugins such as autoformatter